pennylane/devices/execution_config.py

```diff
@@ -76,12 +87,49 @@ def __post_init__(self):
             )
 
         if self.device_options is None:
-            self.device_options = {}
+            # self.device_options = {}
+            object.__setattr__(self, "device_options", {})
```

Comment:
you can use setattr directly like the suggestion. Same for elsewhere in the PR
```suggestion
            setattr(self, "device_options", {})
```
remind me what the problem is with the line you commented?

Reply:
The commented line doesn't work when `frozen=True` for the data class decorator

Reply:
crap. the actual solution to this is to initially assign `device_options` to `field(default_factory=dict)` (which I tried in the original PR). Unfortunately, that's incompatible with Sphinx, meaning the `device_options` attribute doesn't appear in the docs when you do that. This seems the lesser of two evils 👍 nice find!

Reply:
Heads up: I tried your suggested change from `object.__setattr__` to `setattr` and it causes errors regarding the data class being frozen so I've reverted back to `object.__setattr__`